src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java

```diff
@@ -0,0 +1,122 @@
+import org.springframework.security.web.access.AccessDeniedHandler;
+import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@PropertySource("classpath:constants.properties")
+@Configuration
+@EnableWebSecurity
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+    private static final String COOKIES = "JSESSIONID";
+    private final CustomOAuth2UserService oauthUserService;
+    private final UserService userService;
+
+    @Value("${maximum.session.count}")
+    private int MAXIMUM_SESSIONS_COUNT;
+
+    private final UserDetailsService userDetailsService;
+    private final PasswordEncoder encoder;
+
+    public WebSecurityConfig(CustomOAuth2UserService oauthUserService, UserService userService, UserDetailsService userDetailsService, PasswordEncoder encoder) {
+        this.oauthUserService = oauthUserService;
+        this.userService = userService;
+        this.userDetailsService = userDetailsService;
+        this.encoder = encoder;
+    }
+
+    @Bean
+    public AccessDeniedHandler accessDeniedHandler() {
+        return new CustomAccessDeniedHandler();
+    }
+
+    @Bean
+    RestAuthenticationEntryPoint authenticationEntryPoint() {
+        return new RestAuthenticationEntryPoint();
+    }
+
+    @Override
+    protected void configure(final AuthenticationManagerBuilder authenticationManagerBuilder) throws Exception {
+        authenticationManagerBuilder.userDetailsService(userDetailsService).passwordEncoder(encoder);
+    }
+
+    @Override
+    protected void configure(final HttpSecurity http) throws Exception {
+        http
+                .exceptionHandling().accessDeniedHandler(accessDeniedHandler())
+
+                .and()
+                .httpBasic()
+                .authenticationEntryPoint(authenticationEntryPoint())
+
+                .and()
+                .cors()
+                .disable()
+                .csrf()
+                .disable()
+                .authorizeRequests()
+
+                .antMatchers("/").permitAll()
+                .antMatchers("/h2-console/**", "/login**", "/user/registration", "/oauth2**").permitAll()
+
+                .antMatchers("/admin/**")
+                .hasAnyAuthority(Role.ADMIN.name())
+
+                .and()
+                .formLogin().permitAll()
+                .loginPage("/login")
+                .loginProcessingUrl("/perform-login")
+                .usernameParameter("email")
+                .passwordParameter("password")
+                .defaultSuccessUrl("/home")
+                .and()
+                .oauth2Login()
+                .loginPage("/login")
+                .userInfoEndpoint()
+                .userService(oauthUserService)
+                .and()
+                .successHandler(new AuthenticationSuccessHandler() {
```

Comment:
Disgusting! 
Create a separate class, please

Reply:
now in a separate class CustomAuthenticationSuccessHandler